baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferAssertj.java

```diff
@@ -0,0 +1,398 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.ImportTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * {@link PreferAssertj} provides an automated path from legacy test libraries to AssertJ. Our goal is to migrate
+ * existing assertions to AssertJ without losing any information. It's an explicit non-goal to take advantage
+ * of better assertions provided by AssertJ, because those should be implemented in such a way to improve poor
+ * uses of AssertJ as well, which may run after the suggested fixes provided by this checker.
+ *
+ * Remaining work: This doesn't handle Assert.assertThat(value, matcher). We should be able to handle common
+ * cases relatively easily.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "PreferAssertj",
```

Comment:
Do we need to add this to BaselineErrorProneExtension's `DEFAULT_PATCH_CHECKS`  to get the excavator PRs?

Reply:
If we want this applied by default. I noted in the description that I didn't add this to the default patch checks because it doesn't add a dependency on assertj, however the more I think about this, the more I like the idea of running this by default and pushing maintainers to add standard test dependencies to their projects.
Thoughts?

Reply:
I'd just whack it on.

I think in the vast majority of cases it would merge fine because most people have assertj dependencies already... if a repo has a genuine reason why they can't use assertj, they can always turn off the check??

Reply:
Done :-)